daemon/main.go

```diff
@@ -966,11 +969,17 @@ func runDaemon() {
 	// Launch another cilium-health as an endpoint, managed by cilium.
 	log.Info("Launching Cilium health endpoint")
 	if k8s.IsEnabled() {
+
 		// When Cilium starts up in k8s mode, it is guaranteed to be
 		// running inside a new PID namespace which means that existing
 		// PIDfiles are referring to PIDs that may be reused. Clean up.
 		pidfile.Remove(filepath.Join(option.Config.StateDir, health.PidfilePath))
 	}
+
+	// Inject K8s dependency into packages which need to annotate K8s resources.
+	endpoint.EpAnnotator = getEndpointAnnotator()
```

Comment:
did you try `endpoint.EpAnnotator = k8s.Client()` 4 lines above? (inside `k8s.IsEnabled()`)

Reply:
I'll remove these functions in the next upload.